### PR TITLE
Add logic in WriteJSONToResponse to log errors in ECS agent log

### PR DESCRIPTION
**Pull request by chienhanlin**
### Summary
This PR adds logic to propagate responses of requests to the Amazon ECS Container Agent Log when an error occurs.

### Implementation details
This PR includes the following change in agent/handlers/utils/helpers.go:  
- Added logic in `WriteJSONToResponse` function to propagate HTTP response status codes, request types and responses in JSON to the Amazon ECS Container Agent Log when the response status code is 4xx client error or 5xx server error.

### Testing

- Manually tested the changes with following Task Metadata Endpoint (TMDE) version 3 and 4 paths, and get returned JSON responses.

1. ${ECS_CONTAINER_METADATA_URI}
2. ${ECS_CONTAINER_METADATA_URI}/task
3. ${ECS_CONTAINER_METADATA_URI}/stats
4. ${ECS_CONTAINER_METADATA_URI}/task/stats
    
- Manually tested the changes when HTTP response status code 500 occurs, and verified the HTTP response status code, the request type and responses are successfully propagated to Amazon ECS Container Agent Log.
 
- This change is not covered by new tests

### Description for the changelog
Improvement - Propagate responses to the Amazon ECS Container Agent Log when an error occurs

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ubhattacharjya — APPROVED** [review]

File: agent/handlers/utils/helpers.go [hunk 1]
```diff
@@ -129,3 +131,10 @@ func LimitReachedHandler(auditLogger audit.AuditLogger) func(http.ResponseWriter
 		auditLogger.Log(logRequest, http.StatusTooManyRequests, "")
 	}
 }
+
+// WriteErrorToDefaultLogger write the requestType, httpStatusCode and responseJSON of error to the default logger
+func WriteErrorToDefaultLogger(httpStatusCode int, responseJSON []byte, requestType string) {
```
Comment (angelcar) [inline agent/handlers/utils/helpers.go:136]: nit: I think the function name is a bit misleading, there's no error passed as parameter, also the function is not reused anywhere.  I think I'd prefer to get rid of the function and just put the code directly in `WriteJSONToResponse`

File: agent/handlers/utils/helpers.go [hunk 2]
```diff
@@ -129,3 +131,10 @@ func LimitReachedHandler(auditLogger audit.AuditLogger) func(http.ResponseWriter
...
+	if httpStatusCode >= 400 && httpStatusCode <= 599 {
```
Comment (angelcar) [inline agent/handlers/utils/helpers.go:137]: would it be better to have `if httpStatusCode<200 || httpStatusCode>299` ?

**Review by angelcar — APPROVED** [review]

File: agent/handlers/utils/helpers.go (hunk 1 above)
Comment (chienhanlin) [reply agent/handlers/utils/helpers.go]: Thanks for the suggestion. function `WriteErrorToDefaultLogger` is now integrated into function `WriteJSONToResponse`.

File: agent/handlers/utils/helpers.go (hunk 2 above)
Comment (chienhanlin) [reply agent/handlers/utils/helpers.go]: As httpStatusCode 1xx and 3xx would not be considered as an error code; therefore the if statement `httpStatusCode >= 400 && httpStatusCode <= 599` is used here to only propagate httpStatusCode 4xx and 5xx to the agent log.

**Review by angelcar — APPROVED** [review]

**Pull request merged**